Apama current upgrades and future deprecation announcements (#2860)
SoftwareAG/c8y-docs: content/change-logs/analytics/apama-in-c8y-20250129-Upgrade-of-python-java-and-Depreication-of-Support-for-Eclipse-in-Apama.md

---
date: 2025-01-29

> title: Upgrade of python, java and Depreication of Support for Eclipse in Apama

Reviewer: How about ?

```suggestion
title: Upcoming changes to Apama runtime embedded in Streaming Analytics
```

Author: This one looks apt. I will need to change the file name accordingly. 
Also. Should Change the data? 
The last date for change log as you said is 1st Feb Or should I leave it as it is.

Reviewer: no need to change file name or date.

Date can be updated on the day this change get's merged.

change_type:
  - value: change-inv-3bw8e
    label: Announcement
product_area: Analytics
component:
  - value: component-M5-cepIIS
    label: Streaming Analytics
build_artifact:
  - value: tc-KXXmo2SUR
    label: apama-in-c8y
version: 25.339.3
---

> We plan to upgrade Python runtime shipped with Apama to >=3.11 and Java runtime to Java 17. See (latest Apama change logs)[https://cumulocity.com/apama/docs/latest/change-logs] for more details.

Reviewer 2: ```suggestion
The Python runtime shipped with Apama will be upgraded to version >=3.11 and the Java runtime will be upgraded to version Java 17. For details, see (the latest Apama change logs)[https://cumulocity.com/apama/docs/latest/change-logs] in the Apama documentation.
```

> Eclipse and the Windows development support will be removed in 2025 and Python and Java will be upgraded to newer versions. See (latest Apama change logs)[https://cumulocity.com/apama/docs/latest/change-logs] for more details.

Reviewer: ```suggestion
We plan to upgrade Python runtime shipped with Apama to >=3.11 and Java runtime to Java 17. See (latest Apama change logs)[https://cumulocity.com/apama/docs/latest/change-logs] for more details.
```

Author: ### Year seems to be not necessary looks out of place. Year will already be in show prominentley in change log . The rest of the word is good.
Can we mention that it is for Microservice SDK update.

Reviewer: This has nothing to do with the microservice sdk. This is a Java 17 upgrade for Apama.

If year looks odd, just remove the 'In 2025, ' bit

> This will affect the development of Streaming Analytics EPL apps/custom blocks and testing using Pysys locally.

Reviewer: How about ?

```suggestion
The next major releases of Apama will drop support for Windows. It will also mean that Eclipse based development environment will also be unavailable. This will affect the development and testing of EPL apps and Analytics Builder custom blocks natively on Windows.
```

Author: The next major releases of Apama will drop support for Windows and Eclipse. It will imapact the the development and testing of EPL apps and Analytics Builder custom blocks natively on Windows using Eclipse and pysys.

Reviewer: no need to give so much prominence to Eclipse, we are trying to downplay it as we don't have a replacement for now. and no need to talk about pysys (also pysys will work on windows)

> The next major releases of Apama will drop support for Windows. It will also mean that Eclipse based development environment will also be unavailable. This will affect the development and testing of EPL apps and Analytics Builder custom blocks natively on Windows.

Reviewer 2: ```suggestion
The next major releases of Apama will drop support for Windows. As a consequence, the Eclipse-based development environment will also be unavailable. This will affect the development and testing of EPL apps and Analytics Builder custom blocks natively on Windows.
```

---
date: 2025-01-29
title: Upgrade of python, java and Depreication of Support for Eclipse in Apama
change_type:
  - value: change-inv-3bw8e
    label: Announcement
product_area: Analytics
component:
  - value: component-M5-cepIIS
    label: Streaming Analytics
build_artifact:
  - value: tc-KXXmo2SUR
    label: apama-in-c8y
version: 25.339.3
---

> Java version is updated to Java 17 to support the latest Microservice SDK.

Reviewer: when did this happen?

> Java version is updated to Java 17 to support the latest Microservice SDK.

Reviewer: This is not required

Author: deleted it.

> Python version is updated to v3.9.19 

Reviewer: This is not required

Author: removed it.

> Apama support for Eclipse and Windows build will be deprecated in future Apama release. 

Reviewer: ```suggestion
Eclipse and the Windows development support will be removed in 2025 and Python and Java will be upgraded to newer versions. See (latest Apama change logs)[https://cumulocity.com/apama/docs/latest/change-logs] for more details.
```

> This will affect the development of Streaming Analytics EPL apps/custom blocks and testing using Pysys locally.

Reviewer: delete this
